### PR TITLE
rm latest version overwrite for test_iterator

### PR DESCRIPTION
**Pull request by mmsqe**

### CODE REVIEW

File: versiondb/versiondb.py [hunk 1]
```diff
@@ -125,9 +125,6 @@ def iterator(
         start: Optional[bytes] = None,
         reverse: bool = False,
     ):
-        if version is not None and version == self.latest_version():
-            version = None
-
```
Comment (yihuang) [inline versiondb/versiondb.py:130]: why this change?

Comment (mmsqe) [reply versiondb/versiondb.py:130]: query result was reversed and with other store_key, it's to make `test_versiondb.py::test_iterator PASSED`

Comment (yihuang) [reply versiondb/versiondb.py:130]: still don't understand, is this line problematic?

Comment (mmsqe) [reply versiondb/versiondb.py:130]: you could print out the diff
```
expect:  [(b'z-genesis-only', b'2'), (b'modify-in-block2', b'2'), (b'add-in-block2', b'1'), (b'add-in-block1', b'1')]
actual:  [(b'add-in-block1', b'1'), (b'add-in-block2', b'1'), (b'modify-in-block2', b'2'), (b'z-genesis-only', b'2'), (b's/k:staking/key1', b'value2'), (b's/k:staking/key1/subkey', b'value1'), (b's/latest', b'\x08\x04')]
```

Comment (yihuang) [reply versiondb/versiondb.py:130]: the reverse iteration on latest version is problematic, we should fix the root cause

Comment (mmsqe) [reply versiondb/versiondb.py:130]: oh you mean [py](https://github.com/yihuang/python-versiondb/blob/main/versiondb/versiondb.py#L121) not [go](https://github.com/mmsqe/cronos/blob/49f26dbf7938c8a0488791f57c0dbd628c359310/versiondb/tmdb/store.go#L176) version right?

Comment (yihuang) [reply versiondb/versiondb.py:130]: fixed in main branch

Comment (yihuang) [reply versiondb/versiondb.py:130]: > oh you mean [py](https://github.com/yihuang/python-versiondb/blob/main/versiondb/versiondb.py#L121) not [go](https://github.com/mmsqe/cronos/blob/49f26dbf7938c8a0488791f57c0dbd628c359310/versiondb/tmdb/store.go#L176) version right?

yeah, I mean the py version, should probably move the test cases here to the go version, the py test cases are better.
